### PR TITLE
[X86][MC] Teach disassembler to recognize apx instructions which ignores W bit

### PR DESCRIPTION
**Pull request by XinWang10**
Extended VMX instructions and 8 bit apx extended instructions don't need W bit, they are marked as W ignored in spec.
RFC: https://discourse.llvm.org/t/rfc-design-for-apx-feature-egpr-and-ndd-support/73031/4

### CODE REVIEW

File: llvm/lib/Target/X86/X86InstrUtils.td [hunk 1]
```diff
@@ -967,6 +967,7 @@ class ITy<bits<8> o, Format f, X86TypeInfo t, dag outs, dag ins, string m,
       !strconcat(m, "{", t.InstrSuffix, "}\t", args), p>, NoCD8 {
   let hasSideEffects = 0;
   let hasREX_W  = t.HasREX_W;
+  let IgnoresW = !if(!eq(t.VT, i8), 1, 0);
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrUtils.td:970]: “IGNORED” means that the W bit in the EVEX payload is ignored.
Would this change make REX2.W is ignored too?

Comment (XinWang10) [reply llvm/lib/Target/X86/X86InstrUtils.td:970]: Good catch! Let me check and add tests.

Comment (XinWang10) [reply llvm/lib/Target/X86/X86InstrUtils.td:970]: It's ok for REX2.W, 'IGNORED' REX2 instructions would point to the same record whether they have W or not when decoding, these has been handled in current compiler in llvm/utils/TableGen/X86DisassemblerTables.cpp line 100.
`  case IC_64BIT:
    return (inheritsFrom(child, IC_64BIT_REXW) ||`

Comment (KanRobert) [reply llvm/lib/Target/X86/X86InstrUtils.td:970]: Not understood. I think REX.W/REX2.W can not be ignored.

File: llvm/test/MC/Disassembler/X86/apx/IgnoreW.txt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+# ATT:   {evex}	imulb	%bl
+# INTEL: {evex}	imul	bl
+0x62,0xf4,0xfc,0x08,0xf6,0xeb
+
+# ATT:   {nf}	imulb	%bl
+# INTEL: {nf}	imul	bl
+0x62,0xf4,0xfc,0x0c,0xf6,0xeb
+
+# ATT:   imulb	%r16b
+# INTEL: imul	r16b
```
Comment (KanRobert) [inline llvm/test/MC/Disassembler/X86/apx/IgnoreW.txt:117]: What does this test? W bit is 0 here.

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**